src/sig_stfl/sig_stfl.h

```diff
@@ -0,0 +1,659 @@
+ *
+ * @note If the `lock` function and `mutex` are both set, proceed to lock the private key.
+ */
+OQS_STATUS OQS_SIG_STFL_SECRET_KEY_lock(OQS_SIG_STFL_SECRET_KEY *sk);
+
+/**
+ * Unlock the secret key, making it accessible to other processes.
+ *
+ * This function is crucial in environments where multiple processes need to coordinate access to
+ * the secret key, as it allows a process to signal that it has finished using the key, so
+ * others can safely use it.
+ *
+ * @warning If the `unlock` function is set and `mutex` is not set, this unlock operation will fail.
+ *
+ * @param[in] sk Pointer to the secret key whose lock should be released.
+ * @return OQS_SUCCESS if the lock was successfully released; otherwise, OQS_ERROR.
+ *
+ * @note It's not necessary to use this function in either Keygen or Verifying operations.
+ *       In a concurrent environment, the user is responsible for locking and unlocking the private key,
+ *       to make sure that only one thread can access the private key during a Signing operation.
+ *
+ * @note If the `unlock` function and `mutex` are both set, proceed to unlock the private key.
+ */
+OQS_STATUS OQS_SIG_STFL_SECRET_KEY_unlock(OQS_SIG_STFL_SECRET_KEY *sk);
+
+/**
+ * Set the callback and context for securely storing a stateful secret key.
+ *
+ * This function is designed to be called after a new stateful secret key
+ * has been generated. It enables the library to securely store secret key
+ * and update it every time a Signing operation occurs.
+ * Without properly setting this callback and context, signature generation
+ * will not succeed as the updated state of the secret key cannot be preserved.
+ *
+ * @param[in] sk Pointer to the stateful secret key to be managed.
+ * @param[in] store_cb Callback function that handles the secure storage of the key.
+ * @param[in] context Application-specific context that assists in the storage of secret key data.
+ *                    This context is managed by the application, which allocates it, keeps track of it,
+ *                    and deallocates it as necessary.
+ * @return None.
+ */
+OQS_API void OQS_SIG_STFL_SECRET_KEY_SET_store_cb(OQS_SIG_STFL_SECRET_KEY *sk, secure_store_sk store_cb, void *context);
+
+/**
+ * Serialize the stateful secret key data into a byte array.
+ *
+ * Converts an OQS_SIG_STFL_SECRET_KEY object into a byte array for storage or transmission.
+ *
+ * @param[out] sk_buf_ptr Pointer to the allocated byte array containing the serialized key.
+ * @param[out] sk_len Length of the serialized key byte array.
+ * @param[in] sk Pointer to the OQS_SIG_STFL_SECRET_KEY object to be serialized.
+ * @return OQS_SUCCESS on success, or an OQS error code on failure.
+ *
+ * @note The function allocates memory for the byte array, and it is the caller's responsibility to free this memory after use.
+ */
+OQS_API OQS_STATUS OQS_SECRET_KEY_STFL_serialize_key(uint8_t **sk_buf_ptr, size_t *sk_len, const OQS_SIG_STFL_SECRET_KEY *sk);
```

Comment:
Should this actually be
`OQS_SIG_STFL_SECRET_KEY_serialize`?

Reply:
Good catch!